unified-runtime/source/adapters/hip/device.hpp

```diff
@@ -107,11 +111,9 @@ struct ur_device_handle_t_ : ur::hip::handle_base {
 
   int getMaxWorkGroupSize() const noexcept { return MaxWorkGroupSize; };
 
-  int getMaxBlockDimX() const noexcept { return MaxBlockDimX; };
-
-  int getMaxBlockDimY() const noexcept { return MaxBlockDimY; };
+  size_t getMaxBlockDim(int dim) const noexcept { return MaxBlockDim[dim]; };
```

Comment:
Is `noexcept` on something that accesses `MaxBlockDim[dim]` correct? Might'nt this technically throw an exception when out of bounds?

Reply:
It's a `size_t` C array not a vector, so I don't believe this would ever throw an exception. Maybe with extra windows bounds checking or something? But even then I'd expect that to be asserting instead.